daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -338,7 +351,26 @@ private[lf] case class PartialTransaction(
         case None => Right((cid, ptx))
         case Some(kWithM) =>
           val ck = GlobalKey(templateId, kWithM.key)
-          Right((cid, ptx.copy(keys = ptx.keys.updated(ck, Some(cid)))))
+          val globalKeyInputs = keys.get(ck) match {
+            // We saw an archive or negative lookup, no constraint on global key inputs.
+            case Some(None) => ptx.globalKeyInputs
+            // We saw a create or successful lookup, no constraint
+            // on global key inputs.
+            // TODO (MK) In uck-mode this should be an error if there is no archive in between.
+            case Some(Some(_)) => ptx.globalKeyInputs
+            // No entry or archive => key must not be active
+            // at the beginning of the transaction.
```

Comment:
Why not factorizing the two cases ?

Reply:
Mainly because they’re going to be different when we add uniqueness checks and I think it’s useful to already keep them separate for documentation purposes.

Reply:
:+1: